x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/rule_execution_log_client.ts

```diff
@@ -20,17 +20,20 @@ import {
 } from './types';
 
 export interface RuleExecutionLogClientArgs {
+  isRuleRegistryEnabled: boolean;
   ruleDataService: IRuleDataPluginService;
   savedObjectsClient: SavedObjectsClientContract;
 }
 
-const RULE_REGISTRY_LOG_ENABLED = false;
-
 export class RuleExecutionLogClient implements IRuleExecutionLogClient {
   private client: IRuleExecutionLogClient;
 
-  constructor({ ruleDataService, savedObjectsClient }: RuleExecutionLogClientArgs) {
-    if (RULE_REGISTRY_LOG_ENABLED) {
+  constructor({
+    isRuleRegistryEnabled,
+    ruleDataService,
+    savedObjectsClient,
+  }: RuleExecutionLogClientArgs) {
+    if (isRuleRegistryEnabled) {
```

Comment:
Hey @madirey, I'm not sure that we should use Exec Log implementation based on Rule Registry. It has been deprecated, and I'm removing it with [this PR](https://github.com/elastic/kibana/pull/112286/files).

Reply:
@xcrzx Tests were failing because of the missing implementation... I can certainly replace with the new implementation once it's merged.

Reply:
Hmm, that's strange. SO-based implementation should be fully functional. So if the tests are failing, maybe it's a real bug or a false negative? 
I can take a look into it. How can I reproduce the failure?

Reply:
@xcrzx I'll have to look into it, that was many many many test failures ago. 😂   Stand by.

Reply:
Okay, feel free to ping me if you need any help 🙂